### PR TITLE
feat: disable form submit when model is invalid

### PR DESCRIPTION
**Pull request by bombguy**
*Issue #, if available:*

*Description of changes:*
The generated update form submit & clear button should be disabled when the model is invalid

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/codegen-ui/example-schemas/forms/flex-datastore-create.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "MyFlexUpdateForm",
-  "formActionType": "update",
```
Comment (bombguy) [inline packages/codegen-ui/example-schemas/forms/flex-datastore-create.json:3]: Looks like a typo. My change was affecting strange tests.

File: packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react-forms.test.ts.snap [hunk 2]
```diff
@@ -4764,51 +4761,49 @@ export default function MyFlexUpdateForm(props) {
...
     RowGrid0?: FormProps<GridProps>;
     username?: FormProps<TextFieldProps>;
     caption?: FormProps<TextFieldProps>;
     Customtags?: FormProps<TextFieldProps>;
     tags?: FormProps<TextFieldProps>;
     profile_url?: FormProps<TextFieldProps>;
 } & EscapeHatchProps;
-export declare type MyFlexUpdateFormProps = React.PropsWithChildren<{
-    overrides?: MyFlexUpdateFormOverridesProps | undefined | null;
+export declare type MyFlexCreateFormProps = React.PropsWithChildren<{
```
Comment (bombguy) [inline packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react-forms.test.ts.snap:4795]: Diff caused by typo in one of the existing tests.

**Review by lyonsbp — APPROVED** [review]

File: packages/codegen-ui-react/lib/forms/form-renderer-helper.ts [hunk 3]
```diff
@@ -242,54 +242,136 @@ export const addFormAttributes = (component: StudioComponent | StudioComponentCh
...
+      attributes.push(
+        factory.createJsxAttribute(
+          factory.createIdentifier('isDisabled'),
+          factory.createJsxExpression(
+            undefined,
+            factory.createPrefixUnaryExpression(
+              SyntaxKind.ExclamationToken,
+              factory.createParenthesizedExpression(
+                factory.createBinaryExpression(
+                  factory.createIdentifier('id'),
```
Comment (awinberg-aws) [inline packages/codegen-ui-react/lib/forms/form-renderer-helper.ts:255]: Can you help me understand where `id` and `myData` get defined?

Comment (bombguy) [reply packages/codegen-ui-react/lib/forms/form-renderer-helper.ts:255]: Ahhh nice catch! They get both defined at the top of the generated forms. I shouldn't be hardcoding the variable name as it uses the data model's name. Fixing in next rev

File: packages/codegen-ui/lib/utils/form-component-metadata.ts [hunk 4]
```diff
@@ -66,6 +66,7 @@ export const mapFormMetadata = (form: StudioForm, formDefinition: FormDefinition
     id: form.id,
     name: form.name,
     formActionType: form.formActionType,
+    dataType: form.dataType,
```
Comment (bombguy) [inline packages/codegen-ui/lib/utils/form-component-metadata.ts:69]: Exposing data type as form metadata

**Review by lyonsbp — APPROVED** [review]

**Review by awinberg-aws — APPROVED** [review]

**Review by hein-j — APPROVED** [review]

**Pull request merged**